data/pigui/modules/new-game-window/ship.lua

```diff
@@ -735,14 +802,89 @@ function ShipEquip:fromStartVariant(variant)
 		end
 	end
 	self:update()
-	self:setLock(true)
+	self.lock = true
 end
 
 function ShipEquip:isValid()
 	ShipSummary:prepareAndValidateParamList()
 	return ShipSummary.equip.valid
 end
 
+---@param unitBase table where to find the equipment unit
+---@param unitPath string
+---@param eqTable table with IDs, can be subtable
+---@param eqTableSectionPath string
+---@return string? id
+---@return string? errorString
+local function findSavedEquipmentID(unitBase, unitPath, eqTable, eqTableSectionPath)
+	local entry, errorString = Helpers.getByPath(unitBase, unitPath)
+	-- missing entry is acceptable
+	if errorString then return nil end
+	if entry then
+		local eqSection
+		eqSection, errorString = Helpers.getByPath(eqTable, eqTableSectionPath)
+		if errorString then return nil, errorString end
+		assert(eqSection)
+		for id, eq in Helpers.iteratePickled(eqSection) do
+			if eq.ref == entry.ref then return id end
+		end
+	end
+end
+
+ShipEquip.reader = Helpers.versioned {{
+	version = 89,
+	fnc = function(saveGame)
+		-- a table with all possible equipment is located directly in the saveGame
+		-- The ship's equipment set stores refs to this table items, and the
+		-- string IDs of the equipment are only in this table
+		local eqTable, errorString = Helpers.getByPath(saveGame, "lua_modules_json/@Equipment")
```

Comment:
Note: with EquipSet v2 (hopefully this doesn't offend @laarmen - I believe there was only ever 1 EquipSet implementation...), equipment will be migrated to a new "persistent object" feature stored in `lua_modules_persistent`. (It will also need a new frontend for setting equipment when starting the game, but that's a different topic entirely).